### PR TITLE
1.43 SAML doc updates

### PR DESCRIPTION
**Pull request by lyleschemmerling**
Covers
- https://github.com/FusionAuth/fusionauth-issues/issues/1946
- https://github.com/FusionAuth/fusionauth-issues/issues/1900
- https://github.com/FusionAuth/fusionauth-issues/issues/1995
- https://github.com/FusionAuth/fusionauth-issues/issues/2125

All pretty much covered in the saml configuration pages and the API docs by the various new parameters to the configs. They are already in the release notes. For 2125 just added notes around the ACS URL section, not sure much else is needed.

### CODE REVIEW

File: site/docs/v1/tech/apis/identity-providers/_samlv2-request-body.adoc [hunk 1]
```diff
@@ -57,6 +73,20 @@ ifndef::samlv2_idp_initiated[]
...
+Determines if FusionAuth will accept IDP-initiated login requests from this SAMLv2 Identity Provider. A valid license is required to use this feature.
++
+In general FusionAuth does not recommend you utilize this type of login due to the inherent risks of accepting an unsolicited authentication response from a third-party identity provider. In a service provider initiated login, the SAML Authentication Response is returned to the service provider (FusionAuth) in response to an authentication request. This means FusionAuth can validate certain parameters and only accept a response for a request that FusionAuth initiated. In contrast, during an identity provider initiated login, the third-party identity provider sends FusionAuth a SAML Authentication Response indicating a user has been authenticated without any context. The problem is that FusionAuth did not initiate this so it is unexpected, this limits what can be verified and requires FusionAuth to allow a cross site request from this third party. If you must enable this feature to support a legacy integration or because the user experience is of the highest concern, then you may proceed.
```
Comment (robotdan) [inline site/docs/v1/tech/apis/identity-providers/_samlv2-request-body.adoc:80]: We may need some paragraph breaks so it is consumable -it looks like a wall of text when you render the UI.

File: site/docs/v1/tech/apis/identity-providers/_samlv2-response-body.adoc [hunk 2]
```diff
@@ -24,6 +24,22 @@ Determines if a UserRegistration is created for the User automatically or not. I
 [field]#identityProvider.applicationConfiguration``[applicationId]``.enabled# [type]#[Boolean]#::
 Determines if this identity provider is enabled for the Application specified by the `applicationId` key.
 
+ifndef::samlv2_idp_initiated[]
+[field]#identityProvider.assertionConfiguration.destination.alternates# [type]#[Array]# [since]#Available since 1.43.0#::
```
Comment (robotdan) [inline site/docs/v1/tech/apis/identity-providers/_samlv2-response-body.adoc:28]: In the request description you add this:

```
... if the [field]#identityProvider.assertionConfiguration.destination.policy# setting is `AllowAlternates`.
```

Can we use this same description here? Or at a minimum, we should use the proper named value in the API doc which is `AllowAlternates`.

File: site/docs/v1/tech/apis/identity-providers/_samlv2-response-body.adoc [hunk 3]
```diff
@@ -83,11 +109,15 @@ The id of a SAML reconcile lambda that is applied when the identity provider sen
 
 include::_identity-provider-linking-strategy-response-parameter.adoc[]
 
-[field]#identityProvider.loginHintConfiguration.enabled# [type]#[boolean]# [optional]#Optional# [since]#Available since 1.43.0#::
-When enabled and HTTP-Redirect binding are used, a login hint will be passed (when available) as a query parameter to the IdP using the configured parameter name set by the [field]#identityProvider.loginHintConfiguration.parameterName#.
+ifndef::samlv2_idp_initiated[]
+[field]#identityProvider.loginHintConfiguration.enabled# [type]#[Boolean]# [since]#Available since 1.43.0#::
+Determines if FusionAuth will provide the username or email address when available to the IdP as a login hint when HTTP redirect bindings are used to initiate the AuthN request.
```
Comment (robotdan) [inline site/docs/v1/tech/apis/identity-providers/_samlv2-response-body.adoc:114]: Can we use the same description from the request?

File: site/docs/v1/tech/apis/identity-providers/samlv2-idp-initiated.adoc [hunk 4]
```diff
@@ -34,7 +34,24 @@ The following values will likely be required by your SAML v2 IdP Initiated Ident
 
 These values are autogenerated and viewable within the UI after creating the Identity Provider. They can be viewed by navigating to [breadcrumb]#Settings -> Identity Providers -> SAMLv2 IdP Initiated -> View#.
 
-include::docs/v1/tech/identity-providers/_saml-config-urls.adoc[]
+`<base_url>` is the URL for your FusionAuth instance, something like `\https://login.piedpiper.com`.
+
+`<identityProviderId>` is the Id of the Identity Provider, and will be a valid UUID.
+
+`<applicationId>` is the id of the application that is the target of the login, and will be a valid UUID.
```
Comment (robotdan) [inline site/docs/v1/tech/apis/identity-providers/samlv2-idp-initiated.adoc:41]: > ... the Id

File: site/docs/v1/tech/apis/identity-providers/samlv2-idp-initiated.adoc [hunk 5]
```diff
@@ -34,7 +34,24 @@ The following values will likely be required by your SAML v2 IdP Initiated Ident
 
 These values are autogenerated and viewable within the UI after creating the Identity Provider. They can be viewed by navigating to [breadcrumb]#Settings -> Identity Providers -> SAMLv2 IdP Initiated -> View#.
 
-include::docs/v1/tech/identity-providers/_saml-config-urls.adoc[]
+`<base_url>` is the URL for your FusionAuth instance, something like `\https://login.piedpiper.com`.
```
Comment (robotdan) [inline site/docs/v1/tech/apis/identity-providers/samlv2-idp-initiated.adoc:37]: Is this different than what is in the `_saml-config-urls.adoc`?

File: site/docs/v1/tech/identity-providers/_samlv2-integration-details-fields.adoc [hunk 6]
```diff
@@ -3,6 +3,8 @@
 An Assertion Consumer Service (or ACS) is the location at a service provider that accepts messages to establish a session based on the assertion.
 +
 By default the redirect URI will be the first Authorized redirect URI found in the FusionAuth Application OAuth configuration. To specify a redirect URI send the authorized URI in the `RelayState` parameter. Support for using the `RelayState` in this manner is supported in FusionAuth version greater than or equal to `1.41.0`. In prior versions, append a query parameter named `redirect_uri` to the ACS in the IdP configuration.
++
+Note: For IdP initiated logins in order to receive a refresh token the query parameter `?scope=offline_access` will need to be appended to the url. When logging into the FusionAuth Admin application this parameter is included by default and does not need to be supplied.
```
Comment (robotdan) [inline site/docs/v1/tech/identity-providers/_samlv2-integration-details-fields.adoc:7]: This doesn't read quite right.. maybe needs some commas? 

Suggestion:

> Note: To receive a refresh token when completing the OAuth2 workflow when using an IdP initiated login, ensure you request the `offline_access` scope.  To request a scope, add the following query parameter to your configured ACS in your IdP, `?scope=offline_access`. When logging into the FusionAuth admin UI, this step is optional as the `offline_scope` will be implicitly added.

File: site/docs/v1/tech/identity-providers/samlv2/index.adoc [hunk 7]
```diff
@@ -81,6 +81,14 @@ The text to be displayed in the button on the login form. This value defaults to
...
+Allows IdP initiated login requests for this configuration. You must have a valid license to use this feature.
++
+In general FusionAuth does not recommend you utilize this type of login due to the inherent risks of accepting an unsolicited authentication response from a third-party identity provider. In a service provider initiated login, the SAML Authentication Response is returned to the service provider (FusionAuth) in response to an authentication request. This means FusionAuth can validate certain parameters and only accept a response for a request that FusionAuth initiated. In contrast, during an identity provider initiated login, the third-party identity provider sends FusionAuth a SAML Authentication Response indicating a user has been authenticated without any context. The problem is that FusionAuth did not initiate this so it is unexpected, this limits what can be verified and requires FusionAuth to allow a cross site request from this third party. If you must enable this feature to support a legacy integration or because the user experience is of the highest concern, then you may proceed.
```
Comment (robotdan) [inline site/docs/v1/tech/identity-providers/samlv2/index.adoc:87]: Same comment as above. We may want to make an include for this much text.

**Review by robotdan — COMMENTED** [review]
Looks good. Should we add some copy to the IdP initiated SAML v2 IdP to indicate if you want to use both - you no longer have to use the standalone IdP config?

File: site/docs/v1/tech/apis/identity-providers/samlv2-idp-initiated.adoc (hunk 5 above)
Comment (lyleschemmerling) [reply site/docs/v1/tech/apis/identity-providers/samlv2-idp-initiated.adoc:37]: The difference is the ACS urls which can change depending on the idp-init toggle and we need to address with https://github.com/FusionAuth/fusionauth-issues/issues/2146, and when that happens this doc will need updated again as well

File: site/docs/v1/tech/identity-providers/_samlv2-integration-details-fields.adoc (hunk 6 above)
Comment (lyleschemmerling) [reply site/docs/v1/tech/identity-providers/_samlv2-integration-details-fields.adoc:7]: My first pass actually read very similar to this but I tried to cut down the words for more brevity.

File: site/docs/v1/tech/apis/identity-providers/_samlv2-request-body.adoc [hunk 8]
```diff
@@ -27,6 +27,22 @@ Determines if a UserRegistration is created for the User automatically or not. I
...
+endif::[]
+
+ifndef::samlv2_idp_initiated[]
+[field]#identityProvider.assertionConfiguration.destination.policy# [type]#[String]# [optional]#Optional# [default]#defaults to `Enabled`# [since]#Available since 1.43.0#::
+The policy to use when performing a destination assertion on the SAML login request.
++
+The possible values are:
++
+* `Enabled` - Verifies the Destination attribute in the SAML response is equal to the expected Destination which is the FusionAuth ACS (Assertion Consumer Service) URL. This is the default and the recommended setting.
+* `Disabled` - Do not validate the Destination attribute. This is not recommend, please use with caution.
```
Comment (andrewpai) [inline site/docs/v1/tech/apis/identity-providers/_samlv2-request-body.adoc:42]:
```suggestion
* `Disabled` - Do not validate the Destination attribute. This is not recommended, please use with caution.
```

File: site/docs/v1/tech/apis/identity-providers/_samlv2-request-body.adoc [hunk 9]
```diff
@@ -72,11 +102,17 @@ The specified Lambda Id must be of type `SAMLv2Reconcile`.
 
 include::_identity-provider-linking-strategy-request-parameter.adoc[]
 
-[field]#identityProvider.loginHintConfiguration.enabled# [type]#[boolean]# [optional]#Optional# [default]#Defaults to `false`# [since]#Available since 1.43.0#::
-When enabled and HTTP-Redirect binding are used, a login hint will be passed (when available) as a query parameter to the IdP using the configured parameter name set by the [field]#identityProvider.loginHintConfiguration.parameterName#.
+ifndef::samlv2_idp_initiated[]
+[field]#identityProvider.loginHintConfiguration.enabled# [type]#[Boolean]# [optional]#Optional# [default]#defaults to `true`# [since]#Available since 1.43.0#::
+When enabled and HTTP-Redirect binding are used, FusionAuth will provide the username or email address when available to the IdP as a login hint using the configured parameter name set by the [field]#identityProvider.loginHintConfiguration.parameterName# to initiate the AuthN request.
```
Comment (andrewpai) [inline site/docs/v1/tech/apis/identity-providers/_samlv2-request-body.adoc:107]:
```suggestion
When enabled and HTTP-Redirect bindings are used, FusionAuth will provide the username or email address when available to the IdP as a login hint using the configured parameter name set by the [field]#identityProvider.loginHintConfiguration.parameterName# to initiate the AuthN request.
```

File: site/docs/v1/tech/apis/identity-providers/_samlv2-response-body.adoc [hunk 10]
```diff
@@ -24,6 +24,22 @@ Determines if a UserRegistration is created for the User automatically or not. I
...
+endif::[]
+
+ifndef::samlv2_idp_initiated[]
+[field]#identityProvider.assertionConfiguration.destination.policy# [type]#[String]# [since]#Available since 1.43.0#::
+The policy to use when performing a destination assertion on the SAML login request.
++
+The possible values are:
++
+* `Enabled` - Verifies the Destination attribute in the SAML response is equal to the expected Destination which is the FusionAuth ACS (Assertion Consumer Service) URL. This is the default and the recommended setting.
+* `Disabled` - Do not validate the Destination attribute. This is not recommend, please use with caution.
```
Comment (andrewpai) [inline site/docs/v1/tech/apis/identity-providers/_samlv2-response-body.adoc:39]:
```suggestion
* `Disabled` - Do not validate the Destination attribute. This is not recommended, please use with caution.
```

File: site/docs/v1/tech/apis/identity-providers/_samlv2-response-body.adoc [hunk 11]
```diff
@@ -83,11 +109,15 @@ The id of a SAML reconcile lambda that is applied when the identity provider sen
 
 include::_identity-provider-linking-strategy-response-parameter.adoc[]
 
-[field]#identityProvider.loginHintConfiguration.enabled# [type]#[boolean]# [optional]#Optional# [since]#Available since 1.43.0#::
-When enabled and HTTP-Redirect binding are used, a login hint will be passed (when available) as a query parameter to the IdP using the configured parameter name set by the [field]#identityProvider.loginHintConfiguration.parameterName#.
+ifndef::samlv2_idp_initiated[]
+[field]#identityProvider.loginHintConfiguration.enabled# [type]#[Boolean]# [since]#Available since 1.43.0#::
+When enabled and HTTP-Redirect binding are used, FusionAuth will provide the username or email address when available to the IdP as a login hint using the configured parameter name set by the [field]#identityProvider.loginHintConfiguration.parameterName# to initiate the AuthN request.
```
Comment (andrewpai) [inline site/docs/v1/tech/apis/identity-providers/_samlv2-response-body.adoc:114]:
```suggestion
When enabled and HTTP-Redirect bindings are used, FusionAuth will provide the username or email address when available to the IdP as a login hint using the configured parameter name set by the [field]#identityProvider.loginHintConfiguration.parameterName# to initiate the AuthN request.
```

File: site/docs/v1/tech/identity-providers/samlv2/index.adoc [hunk 12]
```diff
@@ -124,6 +132,24 @@ The key used to sign the SAML request. Required when [field]#Sign request# is en
...
+[field]#Login hint parameter name# [optional]#Optional# [since]#Available since 1.43.0#::
+The name of the parameter used to pass the username or email as login hint to the IDP when HTTP redirect bindings are used to initiate the AuthN request. Required when [field]#Enable login hint# is enabled.
+
+[field]#Destination assertion policy# [optional]#Optional# [since]#Available since 1.43.0#::
+The policy to use when performing a destination assertion on the SAML login request.
++
+The possible values are:
++
+* `Enabled` - Verifies the Destination attribute in the SAML response is equal to the expected Destination which is the FusionAuth ACS (Assertion Consumer Service) URL. This is the default and the recommended setting.
+* `Disabled` - Do not validate the Destination attribute. This is not recommend, please use with caution.
```
Comment (andrewpai) [inline site/docs/v1/tech/identity-providers/samlv2/index.adoc:147]:
```suggestion
* `Disabled` - Do not validate the Destination attribute. This is not recommended, please use with caution.
```

File: site/docs/v1/tech/shared/_saml-idp-init-warning.adoc [hunk 13]
```diff
@@ -0,0 +1,5 @@
+In general FusionAuth does not recommend you utilize this type of login due to the inherent risks of accepting an unsolicited authentication response from a third-party identity provider.
++
+In a service provider initiated login, the SAML Authentication Response is returned to the service provider (FusionAuth) in response to an authentication request. This means FusionAuth can validate certain parameters and only accept a response for a request that FusionAuth initiated. In contrast, during an identity provider initiated login, the third-party identity provider sends FusionAuth a SAML Authentication Response indicating a user has been authenticated without any context. The problem is that FusionAuth did not initiate this so it is unexpected, this limits what can be verified and requires FusionAuth to allow a cross site request from this third party.
```
Comment (andrewpai) [inline site/docs/v1/tech/shared/_saml-idp-init-warning.adoc:3]:
```suggestion
In a service provider initiated login, the SAML Authentication Response is returned to the service provider (FusionAuth) in response to an authentication request. This means FusionAuth can validate certain parameters and only accept a response for a request that FusionAuth initiated. In contrast, during an identity provider initiated login, the third-party identity provider sends FusionAuth a SAML Authentication Response indicating a user has been authenticated without any context. The problem is that FusionAuth did not initiate this so it is unexpected, and this limits what can be verified and requires FusionAuth to allow a cross site request from this third party.
```

**Pull request merged**